coderd/workspaceapps.go

```diff
@@ -144,6 +146,37 @@ func (api *API) workspaceAppsProxyPath(rw http.ResponseWriter, r *http.Request)
 		chiPath = "/" + chiPath
 	}
 
+	// Add app usage
+	ctx := r.Context()
+	_, err := api.Database.GetAppUsageByDate(ctx, database.GetAppUsageByDateParams{
+		UserID:     user.ID,
+		TemplateID: workspace.TemplateID,
+		AppID:      app.ID,
+		CreatedAt:  database.Now(),
+	})
+	if err != nil {
+		if !errors.Is(err, sql.ErrNoRows) {
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "Error on getting previous app usage.",
+				Detail:  err.Error(),
+			})
+			return
+		}
+		_, err = api.Database.InsertAppUsage(ctx, database.InsertAppUsageParams{
```

Comment:
Is it possible that there will be a race condition here? If there are 2 HTTP requests submitted at the same time, `api.Database.InsertAppUsage` will be executed twice, right? If so, then we may need a transaction.

Reply:
Ahhh, I see. Inside the transaction, we should check if there is a record and if not, insert it right? That sounds like the right thing to do.